src/Collections-Strings/String.class.st

```diff
@@ -883,7 +895,7 @@ String >> compare: string1 with: string2 collated: order [
 	"('a000' asWideString sameAs: (ByteArray with: 97 with: 0 with: 0 with: 0) asString) >>> false"
 	
 	(string1 isByteString and: [string2 isByteString]) ifTrue: [
-		^ ByteString compare: string1 with: string2 collated: order].
+		^ string1 compareWith: string2 collated: order].
```

Comment:
Doing this change froze everything.

Also, can we blindly do this? I understand that the old primitive returns values 1, 2 and 3 for smaller, equal, bigger. And the second one returns negative, 0 and positive for smaller, equal, bigger. But there is no transformation here between these...

For example, to make this work I had to use instead

```smalltalk
(string1 compareWith: string2 collated: order) sign + 2
```

Am I missing something?

Reply:
Not at all, I just wanted to go incrementally. 
Also, I wasn't aware of the #sign selector, which makes the change very readable. Thanks :)

Also, we are planning to pair program on the issue next week, try out her tools !

Reply:
Sure. My point is: how did you do this change without exploding your image? because as soon as you do this change, nothing else works :)

Reply:
There's not only Image modification :D
emacs String.class.st 
*whistle*